book/src/dev/rfcs/xxxx-zebra-client.md

```diff
@@ -0,0 +1,247 @@
+- Feature Name: (`zebra-client`)
+- Start Date: (2020-10-14)
+- Design PR: [ZcashFoundation/zebra#0000](https://github.com/ZcashFoundation/zebra/pull/1163)
+- Zebra Issue: [ZcashFoundation/zebra#0000](https://github.com/ZcashFoundation/zebra/issues/0000)
+
+# Summary
+[summary]: #summary
+
+The `zebra-client` crate handles *client functionality*. Client functionality
+is defined as all functionality related to a particular user's private data,
+in contrast to the other full node functionality which handles public chain
+state.  This includes:
+
+- note and key management;
+- transaction generation;
+- a client component for `zebrad` that handles block chain scanning, with
+appropriate side-channel protections;
+- an RPC endpoint for `zebrad` that allows access to the client component;
+- a `zebracli` binary that wraps basic wallet functionality and RPC queries in a command-line interface.
```

Comment:
I think we may want to split up `zebracli` even further, and provide a library / binary here as well, where the library provides the core functionality with a nice API and the binary only adds the cli interface ontop of that. That might make it much easier for other developers to leverage the work we did to get a wallet working to add zebra support to their own wallet software.

Reply:
Yep, exactly, that's the intent -- we should express this much more clearly, the idea is for `zebracli` to be a thin wrapper for the `zebra-client` functionality.

Reply:
ooo, so would both binaries, zebrad and zebra-cli be depending on zebra-client?

Reply:
I think the intent was that `zebracli` would be part of the `zebra/client` crate.